src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs

```diff
@@ -3206,6 +3244,10 @@ private static CheckedStatementSyntax TryGetCheckedStatementAncestor(SyntaxNode
             ReportUnmatchedStatements<ForEachStatementSyntax>(diagnostics, match, (int)SyntaxKind.ForEachStatement, oldActiveStatement, newActiveStatement,
                 areEquivalent: AreEquivalentActiveStatements,
                 areSimilar: (n1, n2) => SyntaxFactory.AreEquivalent(n1.Identifier, n2.Identifier));
+
+            ReportUnmatchedStatements<ForEachVariableStatementSyntax>(diagnostics, match, (int)SyntaxKind.ForEachVariableStatement, oldActiveStatement, newActiveStatement,
```

Comment:
>ReportUnmatchedStatements<ForEachVariableStatementSyntax> [](start = 12, length = 57)

We want to match regular ForEach with ForEachVeriable, so I think we need to use 
ReportUnmatchedStatements<CommonForEachStatementSyntax>